libsignify/src/encoding.rs

```diff
@@ -215,20 +215,18 @@ fn read_base64_contents(encoded: &str) -> Result<(Vec<u8>, u64), Error> {
 
     let base64_line = lines.next().ok_or(FormatError::MissingNewline)?;
 
-    if base64_line.is_empty() {
-        return Err(FormatError::LineLength.into());
-    }
-
     let data = base64::decode(base64_line.trim_end()).map_err(|_| FormatError::Base64)?;
 
     match data.get(0..2) {
         // Make sure the specified algorithm matches what we support
         Some(alg) if alg == PKGALG => {
-            // Can't panic, we know there are two lines present.
+            // Find the last newline in the standard encoding scheme. Any bytes after it would be an embedded file.
             let mut newlines = encoded.chars().enumerate().filter(|(_, val)| *val == '\n');
-            let remaining = newlines.nth(1).unwrap().0 as u64;
+            let remaining = newlines.nth(1).ok_or(FormatError::MissingNewline)?.0;
```

Comment:
Isn't this simply

```suggestion
            let remaining = comment_line.len() + base64_line() + 2;
```

?

Reply:
Yeah you're completely right, I made this way too complicated D: Its fixed now but this part of the code honestly embarrasses me that I even pushed it up, so sorry.

Reply:
I could have caught that in the earlier review and I didn't. 
So either we're both embarrassed or we just don't talk about it again ;)

Reply:
works for me either way :)